### PR TITLE
pkg/transport: add 'IsClosedConnError'

### PR DESCRIPTION
**Pull request by gyuho**
Before we do the gRPC graceful shutdown (will return closed listener errors as well),
removing some duplicate code.

### CODE REVIEW

**Comment by gyuho** [comment]
Weird... fails in Go tip with `expect true, got false (accept unix testsocket: use of closed file or network connection)`

File: pkg/transport/listener.go [hunk 1]
```diff
@@ -269,3 +270,27 @@ func ShallowCopyTLSConfig(cfg *tls.Config) *tls.Config {
...
+// IsClosedConnError returns true if the error is from closing listener, cmux.
+// copied from golang.org/x/net/http2/http2.go
+func IsClosedConnError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	// TODO: remove this string search once go/http2 has better type handling
+	operr, ok := err.(*net.OpError)
+	if (ok && operr.Err.Error() == "use of closed network connection") ||
```
Comment (heyitsanthony) [inline pkg/transport/listener.go]: dedup this string into `closedStr := "use of closed network connection"`?

**Comment by gyuho** [comment]
Go tip now returns `internal/poll.ErrClosing` on listener closing. Added match for that.

File: pkg/transport/listener.go [hunk 2]
```diff
@@ -269,3 +270,27 @@ func ShallowCopyTLSConfig(cfg *tls.Config) *tls.Config {
...
+
+	// TODO: match '/go/internal/poll.ErrClosing'
+	if strings.Contains(err.Error(), "use of closed file or network connection") {
+		return true
+	}
+
+	// TODO: match exact error value 'cmux.ErrListenerClosed'
+	// for now just string match without dependency
+	return err.Error() == "mux: listener closed"
+}
```
Comment (heyitsanthony) [inline pkg/transport/listener.go]: would it be enough have:
```go
func IsClosedConnectionError(err error) bool {
    return err != nil && strings.Contains(err.Error(), "closed")
}
```
or is there some case where it has `closed` in the error string but the connection isn't closed?

**Comment by gyuho** [comment]
@heyitsanthony Fixed with comment. PTAL. Thanks

**Pull request merged**